### ISSUE TITLE
fileflares.com [Anti-Adblock]

### ISSUE THREAD

**Issue by KiLL4FuNK** [uBlockOrigin/uAssets#1633]
URL(s) where the issue occurs

`http://fileflares.com/y6ms09353auu`

Describe the issue

Anti-Adblock

Versions

Browser/version: Chrome 62.0.3202.62
uBlock Origin version: 1.15.2

Settings

Default

**Comment by KiLL4FuNK** [comment]
Download

**Comment by KiLL4FuNK** [comment]
Anti-Adblock  @mapx-

**Comment by KiLL4FuNK** [comment]
Anti-Adblock @mapx-

**Comment by KiLL4FuNK** [comment]
Anti-Adblock @mapx-

**Comment by KiLL4FuNK** [comment]
Anti-Adblock

**Comment by ghajini** [comment]
Antiadblock
`fileflares.com/ci84ne4w39yt`
Likely blockadblock

**Comment by ghajini** [comment]
this doesn't fix work for me
after clicking free download ,on next page captcha is hidden by blockadblock dialogue

**Comment by dumbusernameidk** [comment]
@mapx-

Site is now `fflares.com`.

**Comment by mapx-** [comment]
needing test link

**Comment by dumbusernameidk** [comment]
don't have one unfortunately. I stumbled onto the site from google searching. I did try to make an account but the registration kept giving me an error.

**Comment by dumbusernameidk** [comment]
@mapx- 
i'm getting anti-adblock + popups on chrome/firefox. possible to reproduce? the anti-adblock looks like the picture above. this is with the current filters.

**Comment by mapx-** [comment]
maybe they are watching here ..

**Comment by okiehsch** [comment]
I can't reproduce any popups.

**Comment by dumbusernameidk** [comment]
that wouldn't surprise me probably gonna turn into a cat mouse game lol.

also the popup is an oddity seems it only appeared once on both browsers after first loading the page. now it doesn't trigger anymore.

**Comment by okiehsch** [comment]
They forgot `fflares.com##ins.adsbygoogle` but if they are really watching they will close that soon enough, so just add it to your filter list.
The site works fine, the empty space are just an annoyance and frankly one uses such a site to get to the download part as quickly as possible, not to dwell and marvel about the sites layout.

**Comment by okiehsch** [comment]
No, the site validates the timer on the server side

**Comment by dumbusernameidk** [comment]
sometimes it can vary with captchas you either get the ones where you select 3/4 images or the ones where you have to do like 4 pages.

**Comment by dumbusernameidk** [comment]
possible to block all third party scripts for this site? I noticed the site is now loading a bunch of random site names with randomly changing letters. I don't know what the scripts are for but I would assume it's best to block them?

**Comment by mapx-** [comment]
or
`fflares.com^$csp=script-src 'self' 'unsafe-inline' 'unsafe-eval' data: *.google.com *.gstatic.com *.google-analytics.com *.solvemedia.com *.googletagmanager.com`

**Comment by dumbusernameidk** [comment]
@mapx- 

site changed its name back to `fileflares.com`. so these filters need to be updated to include both names:

`fflares.com##+js(abort-on-property-read.js, popjs.init)
@@||google.com/js/*$domain=fflares.com
fflares.com^$csp=script-src 'self' 'unsafe-inline' 'unsafe-eval' data: *.google.com *.gstatic.com *.google-analytics.com *.solvemedia.com *.googletagmanager.com`

**Comment by okiehsch** [comment]
The csp is not needed for that domain, `EasyList` already blocks all third-party requests from `fileflares.com`.

**Comment by KiLL4FuNK** [comment]
Download Free locked

**Comment by dumbusernameidk** [comment]
can confirm site has anti-adblock when clicking free download. test link above still works.

**Comment by mapx-** [comment]
does this work for you
`@@||fileflares.com/$csp,first-party`

?

**Comment by dumbusernameidk** [comment]
Yes that works.

**Comment by KiLL4FuNK** [comment]
I can not advance in free download

**Comment by okiehsch** [comment]
Works for me.
`fflares.com/2qr6cyih5ecs`
![image](https://user-images.githubusercontent.com/16838044/54101937-82e2d480-43c6-11e9-8670-303c8c1b44c9.png)

I can click the "Lib" part of "liberta descarga"

**Comment by dumbusernameidk** [comment]
yeah i have to click the "free" part of the button. for whatever odd reason the whole button isn't clickable just the left side. oddly the premium button works fine...

**Comment by smed79** [comment]
@dumbusernameidk does "free download" button work for you if you disable ubo?

**Comment by dumbusernameidk** [comment]
surprisingly yes.

**Comment by smed79** [comment]
@mapx- please, add
```
||adsco.re^
```

https://vgy.me/PUMcBP.png
https://adtechmadness.wordpress.com/2019/03/09/how-bot-detection-technology-is-abused-by-malvertisers/

**Comment by dumbusernameidk** [comment]
strange i don't get those in my logger. the only 3p xhr request that appears for me is something from `1.2.3.4` which i blocked.

**Comment by ghajini** [comment]
`http://fileflares.com/rrabeqcy79ti`
popup ads when clicking free download,create download link etc

**Comment by okiehsch** [comment]
They are already blocked by `|about:blank|$popup,domain=fileflares.com`.

**Comment by ghajini** [comment]
popup of `amiderations.info`

**Comment by okiehsch** [comment]
I can't reproduce any unblocked popups.

**Comment by smed79** [comment]
![amiderations info](https://user-images.githubusercontent.com/1873139/56062396-24fa3180-5d64-11e9-9396-71e6b4811091.png)

**Comment by okiehsch** [comment]
@smed79 is this popup preceded by an `about:blank` popup?

**Comment by ghajini** [comment]
site updated its code 12/4/2019

**Comment by dumbusernameidk** [comment]
I don't get popups at all for anything on the site.

**Comment by smed79** [comment]
> is this popup preceded by an about:blank popup?

NO.

- https://i.vgy.me/NwBQ64.png
- https://pastebin.com/q3S9SzAM

**Comment by mapx-** [comment]
I get about:blank
Maybe differet page is served to the others, it should work
`*$popup,domain=fileflares.com,3p`

**Comment by okiehsch** [comment]
Well, I can't reproduce the amiderations-popup with Chromiuim or Firefox.

@ghajini if you mean
````
/* Generated 2019-04-12 19:45:02 for "curl%2F7.19.7%20%28x86_64-redhat-linux-gnu%29%20libcurl%2F7.19.7%20NSS%2F3.27.1%20zlib%2F1.2.3%20libidn%2F1.18%20libssh2%2F1.4.2", len 1822 */
(function(){ var d=window;d["\x5f\u0070\u006f\u0070"]=[["\u0073it\x65I\u0064",1510075],
````

That script is the popads script which is blocked already.

**Comment by okiehsch** [comment]
The amiderations-popup is blocked by EasyList looking at **smed79's** logger.

**Comment by ghajini** [comment]
I get intermittent popup of `amiderations.info` like 4/10 ....original tab is closed....

**Comment by dumbusernameidk** [comment]
can't you just add `window.open-defuser.js`? I've had this in my filters for awhile and haven't seen a popup since i added it months back.

**Comment by okiehsch** [comment]
Hm, that does indeed work for me at the moment, I am pretty sure we replaced that because it broke the download button in the past.
@ghajini does `fileflares.com##+js(window.open-defuser.js)` work for your amideration popups too?

**Comment by smed79** [comment]
> fileflares.com##+js(window.open-defuser.js)

work for me + download button is ok (i can download the file without issue)

**Comment by ghajini** [comment]
It prevents popups and allows me to download....so no breakage for me....
Also add `scriptroot` filter for loading bar

**Comment by getworkn** [comment]
Fill recaptcha and submit on 2nd page, 3rd page opened and  download link generated. On 3rd page "Start download" button not working with UBO enabled but working when UBO disabled.

**Comment by okiehsch** [comment]
> Hm, that does indeed work for me at the moment, I am pretty sure we replaced that because it broke the download button in the past.

That is now again the case, will remove the filter.

**Comment by mapx-** [comment]
Can you download on the last page ?
At least in FF `|about:blank|$popup,domain=fileflares.com` avoids for me any download

**Comment by okiehsch** [comment]
Works fine with Chromium on my end.

**Comment by okiehsch** [comment]
I can reproduce with Firefox.
We can add `|about:blank|$popup,domain=fileflares.com,badfilter` for Firefox only, the resulting popup is blocked by an `EasyList`  filter.
It is an ugly solution but defusing the script triggering the popups will also break the download, so I don't see a better way to fix it.

**Comment by mapx-** [comment]
Even whitelisting the popup, when the download should start => I get a new page with "wrong ip" message. Very strange ..

**Comment by okiehsch** [comment]
Have you tried different links, if all of them don't work we should wait before we add the `badfilter`.

**Comment by mapx-** [comment]
The same link `http://fileflares.com/rrabeqcy79ti` is working fine in chrome

**Comment by getworkn** [comment]
FF UBO disabled, download working fine. No message like wrong IP. maybe you are using some kind of vpn / proxy for port 80. they server download via different port than 80

**Comment by okiehsch** [comment]
Yes, the download works fine with Firefox on my end.
@getworkn 
If you add 
````
!#if env_firefox
|about:blank|$popup,domain=fileflares.com,badfilter
!#endif
````
to your filter list does it work without any unblocked popups on your end?

**Comment by getworkn** [comment]
> Yes, the download works fine with Firefox on my end.
> @getworkn
> If you add
> 
> ```
> !#if env_firefox
> |about:blank|$popup,domain=fileflares.com,badfilter
> !#endif
> ```
> to your filter list, does it work without any unblocked popups on your end?

Download is working but opening about:blank page on every page

**Comment by mapx-** [comment]
well, well, ...is working fine using FF stable but not FF dev edition

**Comment by okiehsch** [comment]
That can't be helped, the `about:blank` page should lead to another popup the important question is if that popup is blocked on your end?

**Comment by getworkn** [comment]
> That can't be helped, the `about:blank` page should lead to another popup the important question is if that popup is blocked on your end?

Its ok there is no real ads. blank page is not redirecting to anything.

**Comment by okiehsch** [comment]
Ok, for anybody who is annoyed by the blank page I would suggest you use Chromium/Chrome.

**Comment by dumbusernameidk** [comment]
yeah shame it doesn't work anymore it was working perfectly fine before I suggested the filter. me thinks a certain person is watching the repository. also how scummy do you have to be to try and bypass uBO sheesh.

**Comment by okiehsch** [comment]
> also there's an easylist filter that prevents downloads because of /https?:\/\/.*[=|&|%|#|+].*/$popup,domain

Works fine on my end with Chrome or Firefox.

**Comment by dumbusernameidk** [comment]
> > also there's an easylist filter that prevents downloads because of /https?://._[=|&|%|#|+]._/$popup,domain
> 
> Works fine on my end with Chrome or Firefox.

If the filename has a space in it then the download is blocked because the space is treated as `%`. Can be done with this test link `http://fileflares.com/nsflr8q7gt61`.

**Comment by okiehsch** [comment]
`@@||fileflares.com:$popup,1p`
should fix that.

**Comment by THEtomaso** [comment]
> `@@||fileflares.com:$popup,1p`

Doesn't work with Pale Moon (when I click on "START DOWNLOAD", nothing happens), so I guess a different solution is needed for 'uBlock filters - Legacy'.

--

Also, this download doesn't work for any browsers:
`http://fileflares.com/w4yxt0bqceiu`
After clicking on "FREE DOWNLOAD", we're still met by the same message that ghajini posted a screenshot of here:
https://github.com/uBlockOrigin/uAssets/issues/1633#issuecomment-442819636

**Comment by mapx-** [comment]
for Pale Moon should work this 1
`|about:blank|$popup,domain=fileflares.com,badfilter`

**Comment by THEtomaso** [comment]
> for Pale Moon should work this 1
> `|about:blank|$popup,domain=fileflares.com,badfilter`

Confirmed! :+1:

**Comment by Mennaruuk** [comment]
FileFlares is back at it again:
![image](https://user-images.githubusercontent.com/52135169/107127215-99c82500-6882-11eb-84db-80b427787c05.png)

Seems like this script is the culprit. On each refresh, I got a new number at the end of the URL. For example:

`https://ww7.fileflares.com/js/parking.js?v=1612637389`
`https://ww7.fileflares.com/js/parking.js?v=1612637447`
`https://ww7.fileflares.com/js/parking.js?v=1612637547`

**Comment by Mennaruuk** [comment]
I'm not good with pattern naming. Is this how a filter for this would look like? When I entered it in, it didn't work.

`@@||fileflares.com/js/parking.js?v=*`

**Comment by mapx-** [comment]
`fileflares` is dead (is parked)

**Comment by uBlock-user** [comment]
`||fileflares.com^$,image,redirect-rule=1x1.gif,1p`  you can use that.

**Comment by Mennaruuk** [comment]
Thank you!!
